hoki/csp/tests/test_eventrate.py

```diff
@@ -0,0 +1,180 @@
+"""
+Author: Max Briel
+
+Test for the Complex Stellar Population event rate calculations
+"""
+
+from unittest.mock import patch
+
+import numpy as np
+import pkg_resources
+import pytest
+
+import hoki.csp.eventrate as er
+import hoki.csp.utils as utils
+from hoki.constants import *
+from hoki.csp.sfh import SFH
+from hoki.load import model_output
+from hoki.utils.exceptions import HokiFatalError, HokiFormatError
+
+data_path = pkg_resources.resource_filename('hoki', 'data')
+
+test_sfh = np.loadtxt(f"{data_path}/csp_test_data/mass_points.txt")
+test_metallicity = np.loadtxt(f"{data_path}/csp_test_data/metallicity.txt")
+time_points = np.loadtxt(f"{data_path}/csp_test_data/time_points.txt")
+
+
+# Make test functions from data
+# sfh_fnc = lambda x : np.interp(x, time_points,  test_sfh)
+# Z_fnc = lambda x : np.interp(x, time_points, test_metallicity)
+def sfh_fnc(x): return 1
+def Z_fnc(x): return 0.00001
+
+def vec_sfh(x): return np.zeros(len(x)) +1
+def vec_Z(x): return np.zeros(len(x)) + 0.00001
+
+# define SFH object
+time_axis = np.linspace(0, HOKI_NOW, 1000)
+sfh = SFH(time_axis, "c", {"constant": 1})
+
+
+class TestCSPEventRate():
+
+    data = model_output(f"{data_path}/supernova-bin-imf135_300.zem5.dat")
+
+    # Check initalisation
+    @patch("hoki.load.model_output")
+    def test_init(self, mock_model_output):
+        mock_model_output.return_value = self.data
+        _ = er.CSPEventRate(f"{data_path}", "imf135_300")
+
+    def test_input_functions_at_time(self):
+        assert np.isclose(self.CSP.at_time([sfh_fnc],
+                                                          [Z_fnc],
+                                                          ["Ia"],
+                                                          0,
+                                                          sample_rate=-1)[0]["Ia"],0.002034966495416449),\
+            "Correct input is not taken."
+
+        assert np.isclose(self.CSP.at_time(sfh_fnc,
+                                                          Z_fnc,
```

Comment:
what happened here?

Reply:
If you mean the indenting, that's caused by the change of the function name to a shorter `at_time` instead of `eventrate_at_time`.